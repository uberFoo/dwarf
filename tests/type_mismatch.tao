fn main() {
    let a: int = 3.14;
    let b: float = 42;
    let c: int = "hello";
    let d: string = true;
    let e: bool = 42.0;
}
