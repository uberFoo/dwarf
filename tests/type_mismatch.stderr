[31mError:[0m [1;31merror[0m: Type mismatch: expected `int`, found `float`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:2:12[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m2 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[31mi[0m[31mn[0m[31mt[0m[31m [0m[31m=[0m[38;5;249m [0m[33m3[0m[33m.[0m[33m1[0m[33m4[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m  
 [38;5;240m  â”‚[0m              [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m expected [36mint[0m
 [38;5;240m  â”‚[0m                    [33mâ”‚[0m   
 [38;5;240m  â”‚[0m                    [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m found [36mfloat[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m [1;31merror[0m: Type mismatch: expected `float`, found `int`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:3:12[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m3 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[31mf[0m[31ml[0m[31mo[0m[31ma[0m[31mt[0m[31m [0m[31m=[0m[38;5;249m [0m[33m4[0m[33m2[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m [33mâ”€[0m[33mâ”¬[0m  
 [38;5;240m  â”‚[0m               [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m expected [36mfloat[0m
 [38;5;240m  â”‚[0m                     [33mâ”‚[0m  
 [38;5;240m  â”‚[0m                     [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m found [36mint[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m [1;31merror[0m: Type mismatch: expected `int`, found `String`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:4:12[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m4 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m:[0m[38;5;249m [0m[31mi[0m[31mn[0m[31mt[0m[31m [0m[31m=[0m[38;5;249m [0m[33m"[0m[33mh[0m[33me[0m[33ml[0m[33ml[0m[33mo[0m[33m"[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m [33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m  
 [38;5;240m  â”‚[0m              [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m expected [36mint[0m
 [38;5;240m  â”‚[0m                     [33mâ”‚[0m     
 [38;5;240m  â”‚[0m                     [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m found [36mString[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m [1;31merror[0m: Type mismatch: expected `String`, found `bool`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:5:12[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m5 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249md[0m[38;5;249m:[0m[38;5;249m [0m[31ms[0m[31mt[0m[31mr[0m[31mi[0m[31mn[0m[31mg[0m[31m [0m[31m=[0m[38;5;249m [0m[33mt[0m[33mr[0m[33mu[0m[33me[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m  
 [38;5;240m  â”‚[0m                [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m expected [36mString[0m
 [38;5;240m  â”‚[0m                       [33mâ”‚[0m   
 [38;5;240m  â”‚[0m                       [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m found [36mbool[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m [1;31merror[0m: Type mismatch: expected `bool`, found `float`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:6:12[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m6 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249me[0m[38;5;249m:[0m[38;5;249m [0m[31mb[0m[31mo[0m[31mo[0m[31ml[0m[31m [0m[31m=[0m[38;5;249m [0m[33m4[0m[33m2[0m[33m.[0m[33m0[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m  
 [38;5;240m  â”‚[0m               [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m expected [36mbool[0m
 [38;5;240m  â”‚[0m                     [33mâ”‚[0m   
 [38;5;240m  â”‚[0m                     [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m found [36mfloat[0m
[38;5;246mâ”€â”€â”€â•¯[0m

